### ISSUE TITLE
False positive: "Cannot move" error when dereferencing raw pointers

### ISSUE THREAD

**Issue by oliver-giersch** [intellij-rust/intellij-rust#4795]
## Environment

* **IntelliJ Rust plugin version:** 0.2.112.2148-193
* **Rust toolchain version:** 1.40.0 (73528e339 2019-12-16) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2019.3 (CL-193.5233.103)
* **Operating system:** Linux 5.4.2-1-MANJARO

## Problem description

Reading or writing through a raw pointer results in a "Cannot move" error, which is a false positive in this case.

## Steps to reproduce

```rust
struct Node<T> {
    data: T,
    next: *mut Self,
}

impl<T> Node<T> {
    unsafe fn next(node: *mut Self) -> *mut Self {
        (*node).next // <-- false error here
    }

    unsafe fn set_next(node: *mut Self, next: *mut Self) {
        (*node).next = next; // <-- false error here
    }
}
```
